### DISCUSSION TITLE
Some advice for including pixi.js as dependency in typescript library using rollup?

### DISCUSSION THREAD

**Discussion by kaphula** [pixijs/pixijs#7832] [Q&A]
Hello,

I am trying to build a TypeScript library that can be used directly in browser html file in this manner: 
```html

<!DOCTYPE html>
<html lang="en">
<body>
<script type="module">
    import * as mylib from './mylib.mjs'
    console.log(mylib.sum(2, 2));
</script>
</body>
</html>
```

And I would like to include functionality directly from Pixi to my library. I just can't figure out what kind of settings I should use to make my library work when Pixi is defined as a dependency. If I export functionality that use stuff from Pixi, I get the following error when I run the code above in browser:

```
Uncaught TypeError: Failed to resolve module specifier "url". Relative references must start with either "/", "./", or "../".
```

which I assume is caused by a line that is generated on the top of the built library file:
```
import { parse, format, resolve as resolve$1 } from 'url';
```

I have set up the project configuration [here](https://github.com/kaphula/typescript-rollup-library). I am using rollup to build the library and I would appreciate if somebody could point me to the right direction by showing me similar example projects or instructing what kind of build settings I should use in my `tsconfig.json` and `rollup.config.js` files to make Pixi work in the built library.

**Comment by ShukantPal** [comment]
I think you have to include Node.js built-in polyfills (Rollup has a plugin for this) or not use the `"url"` module.

**Reply by kaphula** [reply]
Hi, thanks for the reply.

I installed rollup-plugin-polyfill-node and added this to my `rollup.config.js`:
```js
nodePolyfills()
// tried this also:
// nodePolyfills( {include: ['url*']} )
```
But with same results as before. Do you know how to **not** use the `"url"` module since it is generated automatically to the built library file? Can it be turned off or switched to something else?
